### DISCUSSION TITLE
possible to add Angular LSP?

### DISCUSSION THREAD

**Discussion by Askath** [zed-industries/zed#13586] [Feature Requests]
Is there any way to include/add the Angular lsp? https://github.com/angular/vscode-ng-language-service
I am not really sure how, as the doc says it is possible but does not further explain how to add it into my config.

**Comment by SuperiorJT** [comment]
From what I have seen, there is an extension being worked on, but one of the blocking issues is that Zed currently sets the LSP based on file extension. Since Angular uses typescript/html file extensions, typescript/html LSP's takes over. Until the Zed team implements a more flexible approach to determine what LSP to use for a given file (in this case, being able to see if the file is in an angular project), we are out of luck.

If you want, you can build the extension and manually import it as a dev extension, but you will have to manually select the LSP each time you open one of those files. The extension can be found here: https://github.com/nathansbradshaw/zed-angular

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
